data-products/src/main/scala/org/sunbird/analytics/audit/AssessmentScoreCorrectionJob.scala

```diff
@@ -0,0 +1,109 @@
+  override def main(config: String)(implicit sc: Option[SparkContext], fc: Option[FrameworkContext]): Unit = {
+    val jobName: String = "AssessmentScoreCorrectionJob"
+    implicit val jobConfig: JobConfig = JSONUtils.deserialize[JobConfig](config)
+    JobLogger.init(jobName)
+    JobLogger.start(s"$jobName started executing", Option(Map("config" -> config, "model" -> jobName)))
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val sc: SparkContext = spark.sparkContext
+    try {
+      spark.setCassandraConf("LMSCluster", CassandraConnectorConf.ConnectionHostParam.option(AppConf.getConfig("sunbird.courses.cluster.host")))
+      val res = CommonUtil.time(processBatches())
+      JobLogger.end(s"$jobName completed execution", "SUCCESS", Option(Map("time_taken" -> res._1, "processed_batches" -> res._2)))
+    } catch {
+      case ex: Exception =>
+        JobLogger.log(ex.getMessage, None, ERROR);
+        JobLogger.end(s"$jobName execution failed", "FAILED", Option(Map("model" -> jobName, "statusMsg" -> ex.getMessage)));
+    }
+    finally {
+      frameworkContext.closeContext()
+      spark.close()
+    }
+  }
+
+  // $COVERAGE-ON$ Enabling scoverage
+  def processBatches()(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sc: SparkContext): List[Map[String, Any]] = {
+    val modelParams = config.modelParams.getOrElse(Map[String, Option[AnyRef]]())
+    val batchIds: List[String] = modelParams.getOrElse("assessment.score.correction.batches", List()).asInstanceOf[List[String]].filter(x => x.nonEmpty)
+    val isDryRunMode = modelParams.getOrElse("isDryRunMode", true).asInstanceOf[Boolean]
+    for (batchId <- batchIds) yield {
+      JobLogger.log("Started Fetching the Incorrect Max Score Value for the Batch", Option(Map("batch_id" -> batchId, "isDryRunMode" -> isDryRunMode)), INFO)
+      removeRecords(batchId = batchId, isDryRunMode = isDryRunMode)
+    }
+  }
+
+  def removeRecords(batchId: String, isDryRunMode: Boolean)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sc: SparkContext) = {
+    val modelParams = config.modelParams.getOrElse(Map[String, Option[AnyRef]]())
+    val outputPath = modelParams.getOrElse("csvPath", "").asInstanceOf[String]
+    // Get the Assessment Data for the specific Batch
+    val assessmentData: DataFrame = getAssessmentAggData(batchId).select("course_id", "batch_id", "content_id", "attempt_id", "user_id", "total_max_score", "total_score").persist()
+    // Take the content Id which is associated to the batch being invoked for the correction
+    val contentId: String = assessmentData.select("content_id").collect().map(_ (0)).toList.head.asInstanceOf[String]
+
+    val contentMetaURL: String = modelParams.getOrElse("contentReadAPI", "https://diksha.gov.in/api/content/v1/read/").asInstanceOf[String]
```

Comment:
This logic assumes that the course has only assessment and that too it is self assess?

Reply:
Yes santhosh